src/rpp/rpp/operators/last.hpp

```diff
@@ -0,0 +1,75 @@
+//                  ReactivePlusPlus library
+//
+//          Copyright Aleksey Loginov 2022 - present.
+//                    TC Wang 2022 - present.
+// Distributed under the Boost Software License, Version 1.0.
+//    (See accompanying file LICENSE_1_0.txt or copy at
+//          https://www.boost.org/LICENSE_1_0.txt)
+//
+// Project home: https://github.com/victimsnino/ReactivePlusPlus
+//
+
+#pragma once
+
+#include <rpp/operators/fwd/last.hpp>
+#include <rpp/operators/take_last.hpp>
+#include <rpp/subscribers/constraints.hpp>
+#include <rpp/utils/exceptions.hpp>
+#include <rpp/utils/functors.hpp>
+#include <rpp/utils/utilities.hpp>
+
+IMPLEMENTATION_FILE(last_tag);
+
+namespace rpp::details
+{
+
+template<constraint::decayed_type Type>
+struct last_state : public take_last_state<Type>
+{
+    explicit last_state() : take_last_state<Type>{1} {}
+};
+
+/**
+ * Functor of last() operator for on_next events.
+ */
+using last_on_next = take_last_on_next;
+
+/**
+ * Functor of last() operator for on_completed event.
+ */
+struct last_on_completed
+{
+    template<constraint::decayed_type Type>
+    void operator()(const constraint::subscriber auto& subscriber,
+                    const last_state<Type>& state) const
+    {
+        auto&& last_value = state.items[0];
```

Comment:
```suggestion
        auto&& last_value = state.items.front();
```

Reply:
> Calling front on an empty container is undefined.

I'd prefer to use `at[0]` with bound checking in case someone changes the size of buffer to 0.